src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c

```diff
@@ -17,495 +242,5248 @@ ActorInit En_Invadepoh_InitVars = {
+    s32 pad;
+    Vec3s* currentPathPoint = &this->pathPoints[this->currentPoint];
+    Vec3s* nextPathPoint = currentPathPoint + 1;
+    Vec3f currentPathPointPos;
+    Vec3f nextPathPointPos;
+
+    if (this->currentPoint != this->endPoint) {
+        Math_Vec3s_ToVec3f(&currentPathPointPos, currentPathPoint);
+        Math_Vec3s_ToVec3f(&nextPathPointPos, nextPathPoint);
+        Math_SmoothStepToS(&this->actor.shape.rot.y, Math_Vec3f_Yaw(&currentPathPointPos, &nextPathPointPos), scale,
+                           step, minStep);
+    }
+}
+
+/**
+ * This function immediately sets the y-rotation of the actor to face the next point in its path.
+ */
+void EnInvadepoh_SetYawAlongPath(EnInvadepoh* this) {
+    s32 pad;
+    Vec3s* currentPathPoint = &this->pathPoints[this->currentPoint];
+    Vec3s* nextPathPoint = currentPathPoint + 1;
+    Vec3f currentPathPointPos;
+    Vec3f nextPathPointPos;
+
+    Math_Vec3s_ToVec3f(&currentPathPointPos, currentPathPoint);
+    Math_Vec3s_ToVec3f(&nextPathPointPos, nextPathPoint);
+    this->actor.shape.rot.y = Math_Vec3f_Yaw(&currentPathPointPos, &nextPathPointPos);
+}
+
+f32 EnInvadepoh_GetTotalPathDistance(EnInvadepoh* this) {
+    s32 pointCount = this->endPoint + 1;
+    s32 i;
+    Vec3f previousPathPointPos;
+    Vec3f currentPathPointPos;
+    Vec3s* currentPathPoint = &this->pathPoints[0];
+    f32 totalPathDistance = 0.0f;
+
+    Math_Vec3s_ToVec3f(&currentPathPointPos, currentPathPoint);
+    currentPathPoint++;
+
+    for (i = 1; i < pointCount; currentPathPoint++, i++) {
+        Math_Vec3f_Copy(&previousPathPointPos, &currentPathPointPos);
+        Math_Vec3s_ToVec3f(&currentPathPointPos, currentPathPoint);
+        totalPathDistance += Math3D_Distance(&previousPathPointPos, &currentPathPointPos);
+    }
+
+    return totalPathDistance;
+}
+
+void EnInvadepoh_InitPath(EnInvadepoh* this, PlayState* play) {
+    Path* path = &play->setupPathList[EN_INVADEPOH_GET_PATH(&this->actor)];
+
+    this->endPoint = path->count - 1;
+    this->pathPoints = Lib_SegmentedToVirtual(path->points);
+}
+
+/**
+ * Sets the actor's current position to be the same as the point along with the path with the index of `point`.
+ */
+void EnInvadepoh_SetPosToPathPoint(EnInvadepoh* this, s32 point) {
+    Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->pathPoints[point]);
+}
+
+/**
+ * Returns true if the dog is considered sufficiently "close enough" to the path it's running along and false otherwise.
+ * Specifically, it checks to see if the dog's position is contained within a parallelogram that looks like this:
+ *           ____________________________
+ *          |                            |
+ *  (a - b) |                            |
+ *          |                            |
+ *          C----------------------------N
+ *          |                            |
+ *  (a - b) |                            |
+ *          |____________________________|
+ *
```

Comment:
When I was researching how this actually worked in practice, I think I convinced myself it was a parallelogram. When I eventually determined it's just a rectangle, I suppose I just left this the way it is, because a rectangle *is* a parallelogram. I can change it, though.